### PR TITLE
[3.11] GH-95736: fix IsolatedAsyncioTestCase to initialize Runner bef…

### PR DESCRIPTION
**Pull request by kumaraditya303**
…ore calling setup functions  (GH-95898)

(cherry picked from commit 9d515997f943b7b510268448f372dabcbf957858)

Co-authored-by: Kumar Aditya <59607654+kumaraditya303@users.noreply.github.com>
Co-authored-by: Serhiy Storchaka <storchaka@gmail.com>

* Issue: gh-95736

### CODE REVIEW

**Review by gvanrossum — APPROVED** [review]
I discussed this with @pablogsal and he says it's okay to backport. He will review and approve when he's seen the code.

**Comment by gvanrossum** [comment]
Marked it as do-not-merge just to give @pablogsal a chance to review before it is merged. Pablo, once you agree that this can be merged into 3.11 please remove that label and approve the PR.

**Review by pablogsal — APPROVED** [review]
LGTM

Thanks for pinging me! 👍

**Pull request merged**
